Sources/SPMClientUtils/ClientUtils.swift

```diff
@@ -0,0 +1,125 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// In this file, only import modules that are part of SwiftPMDataModel (or swift-tools-support-core)!
+// This way, we cannot accidentally use unavailable types in the API intended for clients.
+@_exported import PackageGraph
+@_exported import PackageModel
+@_exported import struct TSCBasic.AbsolutePath
+@_exported import Workspace
+
+// Modules not part of SwiftPMDataModel may be imported with @_implementationOnly.
+@_implementationOnly import Basics
+
+extension Workspace {
+
+    /// Loads a manifest as the root manifest of the workspace.
+    ///
+    /// This method is most easily used with `tsc_await` from the `TSCBasic` module in the `swift-tools-support-core` package:
+    ///
+    /// ```swift
+    /// let manifest = try tsc_await { completion in
+    ///     someWorkspace.loadRootManifest(
+    ///         at: somePath,
+    ///         diagnosticObserver: someObserver,
+    ///         completion: completion
+    ///     )
+    /// }
+    /// ```
+    ///
+    /// - Parameters:
+    ///     - path: The path to the package’s root directory.
+    ///     - diagnosticObserver: An observer to handle diagnostics.
+    ///     - completion: A closure that will be executed when the loading is complete (or has failed).
```

Comment:
What can a client expect about the concurrency aspects of this closure. Are repeated calls serial or overlapping? Do they occur on the main thread, or a background thread, or anywhere?

Reply:
Oops, wrong place.